### ISSUE TITLE
System logs off even if reboot or shutdown are specified

### ISSUE THREAD

**Issue by ladewig** [lithnet/idle-logoff#17]
I  am testing idle-logoff for use in a computer lab, and it works well to have the user logged off after a set amount of time.

If I set the action to reboot or shutdown, however, the system still logs off. I've verified the registry value for "Action" is changing from 0 to 1 or 2, and I've restarted the machine after changing the setting. I've also tried it on both a VM and a physical PC, and I get the same behavior. 

The version installed is 1.1.6999.20776.

**Comment by ryannewington** [comment]
Hi @ladewig,

Do the users have permission to shut down the system? They will be logged off in the event that the Windows shut down system privilege is not available.

Are there any errors in the event log?

Ryan

**Comment by ryannewington** [comment]
@ladewig did you need any additional help with this issue?

**Comment by doomicle** [comment]
I'm having this problem too. Can't get Idlelogoff to reboot the system. It just logs off regardless of what I set.

**Comment by ladewig** [comment]
> @ladewig did you need any additional help with this issue?

Sorry for not getting back sooner. The users who are loggedin do have permission to shutdown or reboot the system. No errors. My only need at the moment is for logging off, so it's working for us, but it would still be useful if we could get it to take the other actions as well.

**Comment by rafaelcarneiromachado** [comment]
I am implementing this amazing tool and faced exactly the same issue, where I'd like to reboot the computer instead of using the standard logoff option.

**Short Version:** Just remove one of the following keys, however this workaround is only applicable if the message you are receiving in the event viewer is: "_**Reboot will not be performed as other sessions are still active. Logging off instead**_"

- **x64: HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Run\lithnet.idlelogoff.exe**
- **x32: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run\lithnet.idlelogoff.exe**


**TL;DR**
While troubleshooting the reason, I found the root cause in my case as the fact the tool is triggering two instances of its process during the logon. The source code contains a condition that checks if there are more than one instance running, to prevent rebooting the machine with another user's session logged-in but disconnected (clever), nonetheless, since the developer uses the native startup feature from Windows, it sets to run the process in both memory spaces, x32 and x64, resulting in two instances running at the same time, preventing the reboot to happen due to the above mentioned condition in the code.

**Comment by ryannewington** [comment]
@rafaelcarneiromachado 

Thanks for the investigation and detailed notes. I'm having trouble reproducing this issue though. I'm wondering if you'd mind helping me troubleshoot this further? I tested on x64 windows 10.

The installer is 32-bit and only, and writes to the 32-bit `Run` key. So on x86 that is `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Run\lithnet.idlelogoff.exe` but on x64 it will be `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Run\lithnet.idlelogoff.exe`

There isn't a case where the installer is configured to write to both locations at once. 

Could the key be left over from a previous install? A base image perhaps?

Ryan

**Comment by rafaelcarneiromachado** [comment]
Hi Ryan,

Maybe this flag?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/platform-compiler-option

Cheers,
Rafael

**Comment by ryannewington** [comment]
Hi @rafaelcarneiromachado,

While the assembly itself it compiled as AnyCPU, the MSI installer, which is the one that creates this key, is x86 only. MSI doesn't have the concept of an anycpu installer. It's either x86 or x64

**Comment by rafaelcarneiromachado** [comment]
It may intercept the installation if you have Registry Redirector enabled:

https://docs.microsoft.com/en-us/windows/win32/winprog64/registry-redirector?redirectedfrom=MSDN

**Comment by ryannewington** [comment]
I found the issue. It only happens if you open the UI and click 'ok'. Leftover code from before the installer created the startup key!

https://github.com/lithnet/idle-logoff/releases/tag/v1.1.7478

Thank you for your help in finding this.

**Comment by rafaelcarneiromachado** [comment]
Thank you for the tool!
